compiler/backend/src/org/jetbrains/kotlin/codegen/StackValue.java

```diff
@@ -640,10 +640,16 @@ public static boolean isUnboxedInlineClass(@NotNull KotlinType kotlinType, @NotN
     }
 
     public static void coerce(@NotNull Type fromType, @NotNull Type toType, @NotNull InstructionAdapter v) {
-        coerce(fromType, toType, v, false);
+        coerce(fromType, toType, v, false, false);
     }
 
-    public static void coerce(@NotNull Type fromType, @NotNull Type toType, @NotNull InstructionAdapter v, boolean forceSelfCast) {
+    public static void coerce(
+            @NotNull Type fromType,
+            @NotNull Type toType,
+            @NotNull InstructionAdapter v,
+            boolean forceSelfCast,
+            boolean isUpcastToJava
```

Comment:
Maybe this parameter should rather be named something like `avoidCheckcast`?

Reply:
Done, I agree it is better.